presto-kudu/src/test/java/io/prestosql/plugin/kudu/TestKuduIntegrationDynamicFilter.java

```diff
@@ -0,0 +1,283 @@
+                        "dynamic-filtering.small-broadcast.max-distinct-values-per-driver", "100",
+                        "dynamic-filtering.small-broadcast.range-row-limit-per-driver", "100",
+                        "dynamic-filtering.large-broadcast.max-distinct-values-per-driver", "100",
+                        "dynamic-filtering.large-broadcast.range-row-limit-per-driver", "100000"),
+                TpchTable.getTables());
+    }
+
+    @AfterClass(alwaysRun = true)
+    public final void destroy()
+    {
+        kuduServer.close();
+    }
+
+    @Test
+    public void testJoinDynamicFilteringNone()
+    {
+        // Probe-side is not scanned at all, due to dynamic filtering:
+        assertDynamicFiltering(
+                "SELECT * FROM lineitem JOIN orders ON lineitem.orderkey = orders.orderkey AND orders.totalprice < 0",
+                withBroadcastJoin(),
+                0,
+                ImmutableSet.of(0));
+    }
+
+    @Test
+    public void testJoinLargeBuildSideDynamicFiltering()
+    {
+        @Language("SQL") String sql = "SELECT * FROM lineitem JOIN orders ON lineitem.orderkey = orders.orderkey and orders.custkey BETWEEN 300 AND 700";
+        int expectedRowCount = 15793;
+        // Probe-side is fully scanned because the build-side is too large for dynamic filtering:
+        assertDynamicFiltering(
+                sql,
+                withBroadcastJoin(),
+                expectedRowCount,
+                ImmutableSet.of(LINEITEM_COUNT, 3977));
+        // Probe-side is partially scanned because we extract min/max from large build-side for dynamic filtering
+        assertDynamicFiltering(
+                sql,
+                withLargeDynamicFilters(),
+                expectedRowCount,
+                ImmutableSet.of(60139, 3977));
+    }
+
+    @Test
+    public void testPartitionedJoinDynamicFiltering()
+    {
+        // Probe-side is not scanned at all
+        assertDynamicFiltering(
+                "SELECT * FROM lineitem JOIN orders ON lineitem.orderkey = orders.orderkey AND orders.totalprice < 0",
+                withPartitionedJoin(),
+                0,
+                ImmutableSet.of(0));
+    }
+
+    @Test
+    public void testJoinDynamicFilteringSingleValue()
+    {
+        // Join lineitem with a single row of orders
+        assertDynamicFiltering(
+                "SELECT * FROM lineitem JOIN orders ON lineitem.orderkey = orders.orderkey AND orders.comment = 'nstructions sleep furiously among '",
+                withBroadcastJoin(),
+                6,
+                ImmutableSet.of(6, 1));
+
+        // Join lineitem with a single row of part
+        assertDynamicFiltering(
+                "SELECT l.comment FROM  lineitem l, part p WHERE p.partkey = l.partkey AND p.comment = 'onic deposits'",
+                withBroadcastJoin(),
+                39,
+                ImmutableSet.of(39, 1));
+    }
+
+    @Test
+    public void testJoinDynamicFilteringImplicitCoercion()
```

Comment:
This case is redundent. Coercion is already tested in different places

Reply:
removed the case